stress: failed test in cockroach/kv/kv.test: TestTxnCoordSenderEndTxn

Binary: cockroach/static-tests.tar.gz sha: https://github.com/cockroachdb/cockroach/commits/c2e0eb023866ede8a42c8a8b4b77cef36825838e

Stress build found a failed test:

```
=== RUN   TestTxnCoordSenderEndTxn
W160508 04:32:25.396138 gossip/gossip.go:892  not connected to cluster; use --join to specify a connected node
I160508 04:32:25.396179 storage/engine/rocksdb.go:137  opening in memory rocksdb instance
I160508 04:32:25.397983 storage/replica_command.go:1452  range 1: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 1s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=1970-01-01 00:00:00 +0000 UTC]
E160508 04:32:25.398208 config/config.go:382  unable to determine largest reserved object ID from system config: descriptor table not found in system config of 0 values
--- FAIL: TestTxnCoordSenderEndTxn (0.01s)
	txn_coord_sender_test.go:444: txn aborted id=d6a8c588 key="key: 2" rw=true pri=0.02945564 iso=SNAPSHOT stat=ABORTED epo=0 ts=0.000000000,95 orig=0.000000000,85 max=0.000000000,85 wto=false
```

Run Details:

```
0 runs so far, 0 failures, over 5s
0 runs so far, 0 failures, over 10s
0 runs so far, 0 failures, over 15s
1 runs so far, 0 failures, over 20s
7 runs so far, 0 failures, over 25s
14 runs so far, 0 failures, over 30s
16 runs so far, 0 failures, over 35s
16 runs so far, 0 failures, over 40s
19 runs so far, 0 failures, over 45s
23 runs so far, 0 failures, over 50s
31 runs so far, 0 failures, over 55s
32 runs so far, 0 failures, over 1m0s
33 runs so far, 0 failures, over 1m5s
36 runs so far, 0 failures, over 1m10s
39 runs so far, 0 failures, over 1m15s
45 runs so far, 0 failures, over 1m20s
49 runs so far, 0 failures, over 1m25s
50 runs so far, 0 failures, over 1m30s
53 runs so far, 0 failures, over 1m35s
56 runs so far, 0 failures, over 1m40s
60 runs completed, 1 failures, over 1m43s
FAIL
```
Please assign, take a look and update the issue accordingly.

--- a/kv/txn_coord_sender_test.go
+++ b/kv/txn_coord_sender_test.go
@@ -299,9 +299,9 @@ func TestTxnCoordSenderHeartbeat(t *testing.T) {
 	var heartbeatTS roachpb.Timestamp
 	for i := 0; i < 3; i++ {
 		util.SucceedsSoon(t, func() error {
-			ok, txn, pErr := getTxn(sender, &initialTxn.Proto)
-			if !ok || pErr != nil {
-				t.Fatalf("got txn: %t: %s", ok, pErr)
+			txn, pErr := getTxn(sender, &initialTxn.Proto)
+			if pErr != nil {
+				t.Fatal(pErr)
 			}
 			// Advance clock by 1ns.
 			// Locking the TxnCoordSender to prevent a data race.
@@ -346,7 +346,7 @@ func TestTxnCoordSenderHeartbeat(t *testing.T) {
 }
 
 // getTxn fetches the requested key and returns the transaction info.
-func getTxn(coord *TxnCoordSender, txn *roachpb.Transaction) (bool, *roachpb.Transaction, *roachpb.Error) {
+func getTxn(coord *TxnCoordSender, txn *roachpb.Transaction) (*roachpb.Transaction, *roachpb.Error) {
 	hb := &roachpb.HeartbeatTxnRequest{
 		Span: roachpb.Span{
 			Key: txn.Key,
@@ -356,9 +356,9 @@ func getTxn(coord *TxnCoordSender, txn *roachpb.Transaction) (bool, *roachpb.Tra
 		Txn: txn,
 	}, hb)
 	if pErr != nil {
-		return false, nil, pErr
+		return nil, pErr
 	}
-	return true, reply.(*roachpb.HeartbeatTxnResponse).Txn, nil
+	return reply.(*roachpb.HeartbeatTxnResponse).Txn, nil
 }
 
 func verifyCleanup(key roachpb.Key, coord *TxnCoordSender, eng engine.Engine, t *testing.T) {
@@ -401,13 +401,20 @@ func TestTxnCoordSenderEndTxn(t *testing.T) {
 		if pErr := txn.Put(key, []byte("value")); pErr != nil {
 			t.Fatal(pErr)
 		}
-		// Conflicting transaction that psuhes the above transaction.
+		// Conflicting transaction that pushes the above transaction.
 		conflictTxn := client.NewTxn(context.Background(), *s.DB)
 		if _, pErr := conflictTxn.Get(key); pErr != nil {
 			t.Fatal(pErr)
 		}
-		// The transaction was pushed to this new timestamp.
-		pushedTimestamp := conflictTxn.Proto.Timestamp.Next().Next()
+
+		// The transaction was pushed at least to conflictTxn's timestamp (but
+		// it could have been pushed more - the push takes a timestamp off the
+		// HLC).
+		pusheeTxn, pErr := getTxn(sender, &txn.Proto)
+		if pErr != nil {
+			t.Fatal(pErr)
+		}
+		pushedTimestamp := pusheeTxn.Timestamp
 
 		{
 			var err error
@@ -433,7 +440,7 @@ func TestTxnCoordSenderEndTxn(t *testing.T) {
 			case 0:
 				// No deadline.
 				if err != nil {
-					t.Error(err)
+					t.Fatal(err)
 				}
 			case 1:
 				// Past deadline.
@@ -441,12 +448,12 @@ func TestTxnCoordSenderEndTxn(t *testing.T) {
 			case 2:
 				// Equal deadline.
 				if err != nil {
-					t.Error(err)
+					t.Fatal(err)
 				}
 			case 3:
 				// Future deadline.
 				if err != nil {
-					t.Error(err)
+					t.Fatal(err)
 				}
 			}
 		}

